src/js/components/AdminPanels.js

```diff
@@ -0,0 +1,299 @@
+import React, { useState, useEffect } from "react"
+import DropDownMenu from "./DropDownMenu"
+import { useModal, ConfirmationModal, ModalProvider } from "./Modal"
+import { simplePost, simpleGet } from "../modules/utils"
+import PropTypes from "prop-types"
+
+// Utility function to fetch and update menus
+const fetchAndUpdateMenus = (redisEndpointUrl, menus, setMenus) => {
+  simpleGet(redisEndpointUrl, { keys: menus.map((menu) => menu.key) }).then(
+    (dataStr) => {
+      const data = JSON.parse(dataStr)
+      const updatedMenus = menus.map((menu) => {
+        const value = data[menu.key]
+        const selectedItem = menu.items.find((item) => item.value === value)
+        return { ...menu, selectedItem }
+      })
+      setMenus(updatedMenus)
+    }
+  )
+}
+
+// Custom hook to handle Redis post success/failure
+function useRedisStatus() {
+  const [redisChanged, setRedisChanged] = useState(null)
+
+  const updateRedisStatus = (status) => {
+    setRedisChanged(status)
+    setTimeout(() => setRedisChanged(null), 2000)
```

Comment:
As discussed internally these setTimeout calls are not really necessary, can you remove this one here and everywhere else :pray: ?